### PR TITLE
React integration

### PR DESCRIPTION
**Pull request by 7KAG7**
PR for react-tifying the index, show and tiles. Also added tests for all.

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -8,6 +8,7 @@ def create
     @fruit = Fruit.find(params[:fruit_id])
     @review = Review.new(review_params)
     @review.fruit = @fruit
+    @review.user_id = current_user.id
```
Comment (prophetsfall) [inline app/controllers/reviews_controller.rb]: you can also just do `@review.user = current_user` thanks to Active Record

File: app/javascript/react/components/FruitShowContainer.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React, { useState, useEffect } from "react"
+import FruitShow from "./FruitShow"
+
+const FruitShowContainer = (props) => {
+  const [fruit, setFruit] = useState({})
+
+  useEffect(() => {
+    const fruitId = props.match.params.id
+
+    fetch(`/api/v1/fruits/${fruitId}`)
```
Comment (prophetsfall) [inline app/javascript/react/components/FruitShowContainer.js]: whenever we issue fetch requests we should always use the error handling.

File: app/models/review.rb [hunk 3]
```diff
@@ -1,5 +1,8 @@
 class Review < ApplicationRecord
-  belongs_to :user
+  validates :rating, presence: true, numericality: {greater_than_or_equal_to: 0, less_than_or_equal_to: 5}
+  validates :body, presence: true
+
   belongs_to :fruit
-  validates :rating, presence: true
+  has_many :users, through: :reviews
+  has_many :countries, through: :fruits_of_countries
 end
```
Comment (prophetsfall) [inline app/models/review.rb]: Your associations are a bit off here. 

A review should belong to both a user and a fruit
if a review `has_many` `:countries, through: :fruits_of_countries` then it also must `has_many :fruits of countries`

File: app/views/fruits/search.html.erb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  <p>No results found.</p>
+  <% else %>
+
+  <p><%= "#{@fruits.count} result(s) found." %></p>
+
+  <ul>
+    <% @fruits.each do |fruit| %>
+    <li>
+      <%= link_to "#{fruit.name} - #{fruit.sci_name}", fruit_path(fruit) %>
+    </li>
```
Comment (prophetsfall) [inline app/views/fruits/search.html.erb]: these three lines should be indented one level (12-14)

File: app/views/fruits/search.html.erb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<div class="search-results">
+  <h1> Search Results </h1>
+
+  <% if !@fruits %>
+  <p>No results found.</p>
+  <% else %>
+
```
Comment (prophetsfall) [inline app/views/fruits/search.html.erb]: this line is extraneous, please remove

Comment (prophetsfall) [reply app/views/fruits/search.html.erb]: Also line 9

File: app/views/fruits/show.html.erb [hunk 6]
```diff
@@ -3,17 +3,31 @@
   <h3><%= @fruit.sci_name %></h3>
   <p><%= @fruit.description %></p>
 
+  <h1>Submit Your Own Review</h1>
+
+<% if current_user %>
```
Comment (prophetsfall) [inline app/views/fruits/show.html.erb]: if a non-logged in user sees this page they will see `Submit Your Own Review` with no form. Consider moving that text within your if

File: app/views/homes/index.html.erb [hunk 7]
```diff
@@ -1,4 +1,3 @@
-<p>
-  Modify me in `app/views/homes/index.html.erb` or change the root directive in
-  `config/routes.rb`
-</p>
+<div id="app">
+
```
Comment (prophetsfall) [inline app/views/homes/index.html.erb]: you don't need the empty line here

File: spec/features/visitor_sees_fruit_spec.rb [hunk 8]
```diff
@@ -1,24 +1,12 @@
-require 'rails_helper'
```
Comment (prophetsfall) [inline spec/features/visitor_sees_fruit_spec.rb]: If this test is no longer used please delete the file.

File: spec/features/visitor_sees_search_bar_spec.rb [hunk 9]
```diff
@@ -0,0 +1,32 @@
+# require 'rails_helper'
```
Comment (prophetsfall) [inline spec/features/visitor_sees_search_bar_spec.rb]: If this test is no longer used please delete the file

File: spec/features/visitor_sees_show_spec.rb [hunk 10]
```diff
@@ -1,13 +1,14 @@
-require 'rails_helper'
```
Comment (prophetsfall) [inline spec/features/visitor_sees_show_spec.rb]: If this test is no longer used, please delete the file

**Comment by prophetsfall** [comment]
Additionally, 34 files changed is too many for one PR. 
Next time I would suggest splitting out the `index` from the `show`

File: app/controllers/reviews_controller.rb (hunk 1 above)
Comment (7KAG7) [reply app/controllers/reviews_controller.rb]: change made

File: app/javascript/react/components/FruitShowContainer.js (hunk 2 above)
Comment (7KAG7) [reply app/javascript/react/components/FruitShowContainer.js]: added error handling

File: app/models/review.rb (hunk 3 above)
Comment (7KAG7) [reply app/models/review.rb]: removed the whole link to the countries b/c we're probably not getting there anyway. Then added the belongs_to :user.

File: app/views/fruits/search.html.erb (hunk 5 above)
Comment (7KAG7) [reply app/views/fruits/search.html.erb]: gone!

File: app/views/fruits/show.html.erb (hunk 6 above)
Comment (7KAG7) [reply app/views/fruits/show.html.erb]: good call

File: app/views/homes/index.html.erb (hunk 7 above)
Comment (7KAG7) [reply app/views/homes/index.html.erb]: gone

File: spec/features/visitor_sees_fruit_spec.rb (hunk 8 above)
Comment (7KAG7) [reply spec/features/visitor_sees_fruit_spec.rb]: done

File: spec/features/visitor_sees_search_bar_spec.rb (hunk 9 above)
Comment (7KAG7) [reply spec/features/visitor_sees_search_bar_spec.rb]: deleted file

File: spec/features/visitor_sees_show_spec.rb (hunk 10 above)
Comment (7KAG7) [reply spec/features/visitor_sees_show_spec.rb]: deleted file

**Comment by 7KAG7** [comment]
> Please see comments and update.
> 
> Also, why did you get rid of your search test?

we react-tified most of these pages and plan to recreate the form in react as well.

File: app/javascript/react/components/FruitShowContainer.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+      .then(response => response.json())
+      .then(fetchedFruit => {
+        setFruit(fetchedFruit)
+      })
+      .catch(error => console.error(`Error in fetch: ${error.message}`))
+    },[])
+
+
+
+
```
Comment (prophetsfall) [inline app/javascript/react/components/FruitShowContainer.js]: I don't know how I missed this last time... you've got a bunch of extra lines here.

**Pull request merged**